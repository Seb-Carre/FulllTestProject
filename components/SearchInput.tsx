import {StyleSheet, TextInput, TextInputProps} from 'react-native';

import React from 'react';

/**
 * Extension of textInput from RN but with a small configuration
 */
export default function SearchInput(props: TextInputProps) {
  return (
    <TextInput style={styles.textInput} placeholder="Search input" {...props} />
  );
}

const styles = StyleSheet.create({
  textInput: {
    marginTop: 24,
    marginBottom: 24,
    marginLeft: 16,
    marginRight: 16,
    padding: 20,
    borderRadius: 5,
    backgroundColor: 'lightgray',
  },
});
