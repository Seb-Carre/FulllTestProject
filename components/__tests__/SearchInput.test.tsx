import {fireEvent, render} from '@testing-library/react-native';

import React from 'react';
import SearchInput from '../SearchInput';

it('render the searchInput as snapShot expected', () => {
  const {toJSON} = render(<SearchInput />);

  expect(toJSON()).toMatchSnapshot();
});

it('render the searchInput and check the placeholder', () => {
  const placeholder = 'Hello';
  const {getByTestId} = render(
    <SearchInput
      testID="searchInput"
      placeholder={placeholder}
      onChangeText={() => {}}
    />,
  );
  const input = getByTestId('searchInput');
  expect(input.props.placeholder).toBe(placeholder);
});

it('render the searchInput and change the text', () => {
  // In this case, we are going to check if our textInput has changed with const text and a mock function which is called if there is a text change as we want to reproduce as the search with GitHub
  const text = '123';
  const mockFunction = jest.fn();
  const {getByTestId} = render(
    <SearchInput testID="searchInput" onChangeText={mockFunction} />,
  );
  const input = getByTestId('searchInput');
  fireEvent.changeText(input, text);
  // By onChangeText, we know if the text has changed !
  expect(mockFunction).toBeCalled();
});
