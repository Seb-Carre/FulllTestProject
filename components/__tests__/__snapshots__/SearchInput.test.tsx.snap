// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the searchInput as snapShot expected 1`] = `
<TextInput
  placeholder="Search input"
  style={
    {
      "backgroundColor": "lightgray",
      "borderRadius": 5,
      "marginBottom": 24,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 24,
      "padding": 20,
    }
  }
/>
`;
